bletransportsample/src/main/java/com/ledger/live/bletransportsample/screen/scan/ScanViewModel.kt

```diff
@@ -0,0 +1,160 @@
+                    devices = state.scannedDevices.map { UiDevice(it.id, it.name) }
+                )
+                _uiState.tryEmit(scanningState)
+            }
+   /*         is BleState.Connected -> {
+                Timber.d("connected")
+                val device = state.connectedDevice
+                _uiState.tryEmit(BleUiState.Connected(UiDevice(device.id, device.name)))
+            }*/
+            else -> {}
+        }
+    }
+
+    fun toggleScan() {
+        if (isScanning) {
+            isScanning = false
+            bleManager.stopScanning()
+            _uiState.tryEmit(BleUiState.Idle)
+        } else {
+            isScanning = true
+            bleManager.startScanning {
+                val scanningState = BleUiState.Scanning(
+                    devices = it.map { device -> UiDevice(device.id, device.name) }
+                )
+                _uiState.tryEmit(scanningState)
+            }
+
+            _uiState.tryEmit(
+                BleUiState.Scanning(
+                    devices = emptyList()
+                )
+            )
+        }
+    }
+
+    @SuppressLint("MissingPermission")
+    fun connectToDevice(address: String) {
+        bleManager.connect(
+            address = address,
+            onConnectError = {
+                Timber.e("Disconnect Callback => cause: $it")
+                isScanning = false
+                _uiState.tryEmit(BleUiState.Idle)
+            },
+            onConnectSuccess = {
+                Timber.d("Connected Callback")
+                isScanning = false
+                val device = it
+                _uiState.tryEmit(BleUiState.Connected(UiDevice(device.id, device.name)))
+            }
+        )
+    }
+
+    fun connectNanoX() {
+        connectToDevice("DE:F1:55:51:C2:0C")
+    }
+
+    fun sendSmallApdu() {
+        Timber.d("Send Small APDU")
+        val hexStringApdu = "b001000000"
+        bleManager.send(
+            onSuccess = {
+                Timber.d("Send Apdu Success Callback : $it")
+
+            },
+            onError = { message ->
+                Timber.e(message)
+            },
+            apduHex = hexStringApdu,
+        )
+    }
+
+    fun sendBigApdu() {
+        Timber.d("Send Big APDU")
+        val hexStringApdu =
+            "000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
+        bleManager.send(
+            apduHex = hexStringApdu,
+            onError = {
+                Timber.e(it)
+            },
+            onSuccess = {
+                Timber.d("Send Apdu Success Callback")
+            }
+        )
+    }
+
+    fun disconnect() {
+        bleManager.disconnect()
+        bleManager.disconnect {
```

Comment:
[ASK] Do we need two methods : `bleManager.disconnect()` and `bleManager.disconnect {...}`

Reply:
It was a test to check that 2 consecutive disconnect was not crashing ^^'